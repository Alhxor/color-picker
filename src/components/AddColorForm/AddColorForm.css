form {
    min-width: 50%;
}

fieldset {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 1em auto;
    border: 1px dashed hsla(0, 0%, 0%, 0.4);
    min-width: 20em;
    min-height: 3em;
}

input {
    display: inline-block;
    padding: 0 .4em;
    margin: .25em .2em 0 0;
    font: inherit;
    vertical-align: middle;
    height: 1.5em;
}

input[type="text"] {
    width: 5em;
}

input[type="color"] {
    width: 3em;
}
