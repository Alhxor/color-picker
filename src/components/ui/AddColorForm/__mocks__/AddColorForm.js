const AddColorFormMock = () => (
    <form></form>
)

AddColorFormMock.displayName = "AddColorFormMock";
export default AddColorFormMock;
