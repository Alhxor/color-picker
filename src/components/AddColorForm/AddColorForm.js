import React from 'react';
import PropTypes from 'prop-types';
import './AddColorForm.css';

class AddColorForm extends React.Component {
    constructor(props) {
        super(props);
        this.onNewColor = props.onNewColor;
        this.state = {
            title: '',
            hexCode: '#000000',
        };
    }

    onChangeInput = stateKey => ({ target }) => {
        this.setState({
            [stateKey]: target.value,
        });
    }

    onSubmit = (event) => {
        event.preventDefault();
        const { title, hexCode } = this.state;
        this.onNewColor({ title, hexCode });
    }

    render = () => (
        <form onSubmit={this.onSubmit}>
            <fieldset>
                <legend>Add a new color</legend>
                <div className="formElementsContainer">
                    <input type="text" placeholder="Color title"
                        onChange={this.onChangeInput('title')} />
                    <input type="color" onChange={this.onChangeInput('hexCode')} />
                    <input type="submit" value="Add" />
                </div>
            </fieldset>
        </form>
    );
}

AddColorForm.propTypes = {
    onNewColor: PropTypes.func.isRequired,
};

export default AddColorForm;
