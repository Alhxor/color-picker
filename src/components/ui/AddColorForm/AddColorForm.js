import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './AddColorForm.css';

class AddColorForm extends Component {
    constructor() {
        super();
        this.state = {
            title: "",
            hexCode: "#000000",
        };
    }

    updateInput = stateKey => ({ target }) => {
        this.setState({
            [stateKey]: target.value,
        });
    }

    onSubmit = (event) => {
        event.preventDefault();
        const { title, hexCode } = this.state;
        this.props.addColor(title, hexCode);
    }

    render = () => (
        <form onSubmit={this.onSubmit}>
            <fieldset>
                <legend>Add a new color</legend>
                <div className="formElementsContainer">
                    <input type="text" placeholder="Color title"
                        onChange={this.updateInput('title')} />
                    <input type="color"
                        onChange={this.updateInput('hexCode')} />
                    <input type="submit" value="Add" />
                </div>
            </fieldset>
        </form>
    );
}

AddColorForm.propTypes = {
    addColor: PropTypes.func.isRequired,
};

export default AddColorForm;
