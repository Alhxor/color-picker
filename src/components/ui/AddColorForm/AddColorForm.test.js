import AddColorForm from './AddColorForm';
import { mount } from 'enzyme';

describe("<AddColorForm />", () => {
    let wrapper;
    const addColor = jest.fn();
    beforeAll(() => {
        wrapper = mount(<AddColorForm addColor={addColor} />);
    });

    it("renders without crashing", () =>
        expect(wrapper.exists()).toBeTruthy());

    it("calls addColor when clicking submit", () => {
        wrapper.find("form").simulate("submit");
        expect(addColor).toHaveBeenCalled();
    });

    it("matches snapshot", () =>
        expect(wrapper).toMatchSnapshot());
});
