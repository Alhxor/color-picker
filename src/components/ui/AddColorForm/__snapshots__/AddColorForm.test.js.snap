// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddColorForm /> matches snapshot 1`] = `
<AddColorForm
  addColor={
    [MockFunction] {
      "calls": Array [
        Array [
          "",
          "#000000",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Add a new color
      </legend>
      <div
        className="formElementsContainer"
      >
        <input
          onChange={[Function]}
          placeholder="Color title"
          type="text"
        />
        <input
          onChange={[Function]}
          type="color"
        />
        <input
          type="submit"
          value="Add"
        />
      </div>
    </fieldset>
  </form>
</AddColorForm>
`;
